src/shared/components/query/QueryStore.ts

```diff
@@ -124,6 +128,13 @@ export class QueryStore
 		this._selectedStudyIds = this.forDownloadTab ? ids.slice(-1) : ids;
 	}
 
+	@computed get selectedStudyIdsSet():{[studyId:string]:boolean}
+	{
+		return this.selectedStudyIds.reduce((ret:{[studyId:string]:boolean}, next:string)=>{

```

Comment:
Might be easier to read if you give this anonymous function a simple name like `setStudyIdTrue` or something like that.